plugin/trino-delta-lake/src/test/java/io/trino/plugin/deltalake/BaseDeltaLakeRegisterTableProcedureTest.java

```diff
@@ -0,0 +1,319 @@
+                ".*Table already exists: '(.*)'.*");
+        assertUpdate("DROP TABLE " + tableName);
+    }
+
+    @Test
+    public void testRegisterTableWithInvalidUriScheme()
+    {
+        String tableName = "test_register_table_with_invalid_uri_scheme_" + randomNameSuffix();
+        String tableLocation = "invalid://hadoop-master:9000/test/delta-lake/hive/orders_5-581fad8517934af6be1857a903559d44";
+        assertQueryFails(format("CALL system.register_table('%s', '%s', '%s')", SCHEMA, tableName, tableLocation),
+                ".*Failed checking table location (.*)");
+    }
+
+    @Test
+    public void testRegisterTableWithInvalidParameter()
+    {
+        String tableName = "test_register_table_with_invalid_parameter_" + randomNameSuffix();
+        String tableLocation = "/test/delta-lake/hive/table1/";
+
+        assertQueryFails(format("CALL system.register_table('%s', '%s')", SCHEMA, tableName),
+                ".*'TABLE_LOCATION' is missing.*");
+        assertQueryFails(format("CALL system.register_table('%s')", SCHEMA),
+                ".*'TABLE_NAME' is missing.*");
+        assertQueryFails("CALL system.register_table()",
+                ".*'SCHEMA_NAME' is missing.*");
+
+        assertQueryFails(format("CALL system.register_table(NULL, '%s', '%s')", tableName, tableLocation),
+                ".*schema_name cannot be null or empty.*");
+        assertQueryFails(format("CALL system.register_table('%s', NULL, '%s')", SCHEMA, tableLocation),
+                ".*table_name cannot be null or empty.*");
+        assertQueryFails(format("CALL system.register_table('%s', '%s', NULL)", SCHEMA, tableName),
+                ".*table_location cannot be null or empty.*");
+
+        assertQueryFails(format("CALL system.register_table('', '%s', '%s')", tableName, tableLocation),
+                ".*schema_name cannot be null or empty.*");
+        assertQueryFails(format("CALL system.register_table('%s', '', '%s')", SCHEMA, tableLocation),
+                ".*table_name cannot be null or empty.*");
+        assertQueryFails(format("CALL system.register_table('%s', '%s', '')", SCHEMA, tableName),
+                ".*table_location cannot be null or empty.*");
+    }
+
+    protected String getTableLocation(String tableName)
+    {
+        Pattern locationPattern = Pattern.compile(".*location = '(.*?)'.*", Pattern.DOTALL);
+        Matcher m = locationPattern.matcher((String) computeActual("SHOW CREATE TABLE " + tableName).getOnlyValue());
+        if (m.find()) {
+            String location = m.group(1);
+            verify(!m.find(), "Unexpected second match");
+            return location;
+        }
+        throw new IllegalStateException("Location not found in SHOW CREATE TABLE result");
+    }
+
+    protected void dropTableFromMetastore(String tableName)
+    {
+        metastore.dropTable(SCHEMA, tableName, false);
```

Comment:
please inline this call - we're not supposed to make sure that `metastore.dropTable` works as expected while testing `register_table`.